packages/syft/src/syft/service/worker/worker_pool_service.py

```diff
@@ -195,10 +184,21 @@ def get_worker(
 
         worker_pool, worker = worker_pool_worker
 
-        worker_status = _get_worker_container_status(worker)
+        with contextlib.closing(docker.from_env()) as client:
+            container = _get_worker_container(client, worker)
```

Comment:
@kiendang  I think it would be good to have a separate function utility function for retrieving the worker status.
Because I saw in Khoa's PR we used utility funciton in displaying statuses.
I will defer the decision to you post merge which we could implement in Khoa's PR

Reply:
As I explained here https://github.com/OpenMined/PySyft/pull/8347#discussion_r1429890863 I don't think a function like `def _get_worker_container_status(worker: Worker) -> WorkerStatus` would be good. We need to get the `Container` from the `worker` first and we might need to use the `Container` for something else beside getting the status, so this would be better `def _get_worker_container_status(container: Container) -> WorkerStatus` and this is not much different from just having a mapping `_CONTAINER_STATUS_TO_WORKER_STATUS: dict[str, WorkerStatus]` anyway and call `_CONTAINER_STATUS_TO_WORKER_STATUS.get(worker_container.status)`. We could wrap that in a function but it doesn't necessarily simplify things.

Reply:
On second thoughts, it's not a bad idea to wrap everything in a function, just need to take notes whether we've already got the worker container object. Will make changes in Khoa's PR.